src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
```

Comment:
The assumption that the separator is always at index 3 of the `fromFormat` array may not be valid for all date formats. Consider dynamically determining the separator by checking for non-alphanumeric characters in the format string.